app/battle.py

```diff
@@ -0,0 +1,15 @@
+from typing import Dict
+from app.knight import Knight
+
+
+def battle_between(knight1: Knight, knight2: Knight) -> Dict[str, int]:
+    knight1.prepare_for_battle()
+    knight2.prepare_for_battle()
+
+    knight1.take_damage(knight2.power)
+    knight2.take_damage(knight1.power)
```

Comment:
The current implementation of `take_damage` does not consider the knight's protection. According to the task description, the damage should be calculated as `enemy_power - self.protection`. You should modify the `take_damage` method in the `Knight` class to reflect this logic.